plugins/module_utils/ece.py

```diff
@@ -110,39 +174,186 @@ def get_snapshot_repo_by_name(self, repo_name):
+              'repository_name': self.snapshot_settings['repository_name']
+            }
+          },
+          'enabled': self.snapshot_settings['enabled'],
+          'retention': {
+            'snapshots': self.snapshot_settings['snapshots_to_retain'],
+          },
+          'interval': self.snapshot_settings['snapshot_interval']
+        }
+        x = 0
+        for resource in data['resources']['elasticsearch']:
+          if resource['ref_id'] == "main-elasticsearch":
+            if not 'settings' in data['resources']['elasticsearch'][x]:
+              data['resources']['elasticsearch'][x]['settings'] = {}
+            data['resources']['elasticsearch'][x]['settings']['snapshot'] = snapshot_settings
+          x = x + 1
+
+      if self.traffic_rulesets:
+        ip_filtering_data = {
+          'rulesets': [self.get_traffic_ruleset_by_name(x)['id'] for x in self.traffic_rulesets]
+        }
+
+      endpoint = 'deployments'
+      cluster_creation_result = self.send_api_request(endpoint, 'POST', data=data)
+      if self.wait_for_completion:
+        elastic_deployment_result = self.wait_for_cluster_state(cluster_creation_result['id'],'elasticsearch','main-elasticsearch','started', self.completion_timeout)
+        kibana_deployment_result = self.wait_for_cluster_state(cluster_creation_result['id'],'kibana','main-kibana','started', self.completion_timeout)
+        if not elastic_deployment_result and not kibana_deployment_result:
+          return False
+      return cluster_creation_result
+
+  def get_matching_clusters(self):
+    clusters = self.get_clusters_by_name(self.cluster_name)
+    return clusters
+
+  def delete_cluster(self, deployment_id, resource_kind = "elasticsearch"):
+    self.terminate_cluster(deployment_id)
+    #endpoint = f'clusters/elasticsearch/{cluster_id}'
+    #target_resource_object = self.get_deployment_resource_by_kind(deployment_id, resource_kind)
+    #target_resource_ref_id = target_resource_object['ref_id']
+    #endpoint = f'deployments/{deployment_id}/{resource_kind}/{target_resource_ref_id}'
+    endpoint = f'deployments/{deployment_id}'
+    delete_result = self.send_api_request(endpoint, 'DELETE')
+    return delete_result
+
+  def terminate_cluster(self, deployment_id, resource_kind = "elasticsearch"):
+    #endpoint = f'clusters/elasticsearch/{cluster_id}/_shutdown'
+    target_deployment_object = self.get_deployment_byid(deployment_id)
+    resources = target_deployment_object['resources'][resource_kind]
+    for resource in resources: 
+      target_resource_ref_id = resource['ref_id']
+      endpoint = f'deployments/{deployment_id}/{resource_kind}/{target_resource_ref_id}/_shutdown'
+      stop_result = self.send_api_request(endpoint, 'POST')
+      wait_result = self.wait_for_cluster_state(deployment_id, resource_kind, target_resource_ref_id,'stopped', self.completion_timeout)
+    if not wait_result:
+      self.module.fail_json(msg=f'failed to stop cluster {self.cluster_name}')
```

Comment:
Am I missing something or would this mean that any time you set `wait_result` to false the module would return a failure, regardless of the success or failure of the command?

Reply:
The wait_result value  is a result of the wait_for_cluster_state.  I assume the intention was to wait for 600 seconds by default and if it still doesn't stop to return a failure.  I didn't really create that part of the code I was just trying to get the status returned properly. If you want it to behave differently, let me know what you want the it to be. I believe this is true for creating the elastic instance as well

Reply:
ohh yea I was reading the code wrong, I see what that's doing now.  looks good to me